Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java

```diff
@@ -0,0 +1,261 @@
+        synchronized (players) {
+            PlayerLeftSessionMessage message = new PlayerLeftSessionMessage(player.getPlayerId());
+
+            players.remove(player);
+            players.stream()
+                    .map(PlayerWrapper::getConnection)
+                    .forEach(c -> c.send(message));
+        }
+
+        Player unwrapped = player.unwrap();
+        unwrapped.setCurrentSession(null);
+        unwrapped.setPlayerId(0);
+
+        synchronized (teams) {
+            Team team = teams.remove(player);
+            if (team != null) {
+                team.leave(unwrapped);
+            }
+        }
+
+        String roomName = room.getName();
+        LOGGER.log(Level.INFO, "Player {0} left session {1}.", new String[]{player.getUsername(), roomName});
+
+        if (room.getOccupancy() <= 0) {
+            GameServer.getInstance().getSessionController().destroySession(room.getName());
+            return;
+        }
+
+        GameServerRmiController rmiController = GameServer.getInstance().getRmiControler();
+        rmiController.decreaseOccupancy(roomName);
+        rmiController.changeHost(roomName, room.getHost().getUsername());
+    }
+
+    public void switchTeam(PlayerWrapper player) {
+        synchronized (teams) {
+            Team oldTeam = teams.remove(player);
+            Team newTeam = oldTeam.getTeamColour() == TeamColour.BLUE ? room.getTeamRed() : room.getTeamBlue();
+
+            teams.put(player, newTeam);
+
+            Player unwrapped = player.unwrap();
+            oldTeam.leave(unwrapped);
+            newTeam.join(unwrapped);
+
+            SwitchTeamMessage message = new SwitchTeamMessage(player.getPlayerId(), newTeam.getTeamColour());
+            players.stream()
+                    .map(PlayerWrapper::getConnection)
+                    .forEach(c -> c.send(message));
+        }
+    }
+
+    public void sendChatMessage(PlayerWrapper player, String message) {
+        if (message.startsWith("/")) {
+            String[] components = message.substring(1).split(" ");
+            String name = components[0];
+
+            String[] arguments = new String[components.length - 1];
+            System.arraycopy(components, 1, arguments, 0, arguments.length);
+
+            Command command = new Command(name, arguments);
+            CommandDispatcher.dispatch(player, command);
+            return;
+        }
+
+        if (mutedNames.contains(player.getUsername().toLowerCase())) {
+            player.getConnection().send(new ChatMessage(-1, Privilege.ADMINISTRATOR, "You have been muted and, therefor, cannot talk."));
+            return;
+        }
+
+        ChatMessage m = new ChatMessage(player.getPlayerId(), player.getPrivilege(), message);
+        synchronized (players) {
+            players.stream()
+                    .map(PlayerWrapper::getConnection)
+                    .forEach(c -> c.send(m));
+        }
+    }
+
+    public void mutePlayer(PlayerWrapper issuer, String username) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)